app/src/main/java/org/greenstand/android/TreeTracker/permission/Permissions.kt

```diff
@@ -0,0 +1,174 @@
+                                Text(
+                                    text = stringResource(R.string.accept_location_permission_message),
+                                    color = Color.Green
+                                )
+                            },
+                            buttons = {
+                                Button(
+                                    modifier = Modifier.wrapContentSize(),
+                                    onClick = {
+                                        perm.launchPermissionRequest()
+                                        navController.popBackStack()
+                                    }
+                                ) {
+                                    Text(
+                                        text = stringResource(R.string.accept_permission)
+                                    )
+                                }
+                            })
+                    }
+
+                }
+            }
+            Manifest.permission.ACCESS_COARSE_LOCATION -> {
+                when {
+                    perm.hasPermission -> {
+                        // TODO Check if location is enabled from the LocationUpdateManager before calling enableLocation
+                        return
+                    }
+                    (perm.shouldShowRationale || !perm.permissionRequested) -> {
+                        AlertDialog(
+                            onDismissRequest = { navController.popBackStack() },
+                            title = {
+                                Text(text = stringResource(R.string.accept_location_permission_header))
+                            },
+                            text = {
+                                Text(
+                                    text = stringResource(R.string.accept_location_permission_message),
+                                    color = Color.Green
+                                )
+                            },
+                            buttons = {
+                                Button(
+                                    modifier = Modifier.wrapContentSize(),
+                                    onClick = {
+                                        perm.launchPermissionRequest()
+                                        navController.popBackStack()
+                                    }
+                                ) {
+                                    Text(
+                                        text = stringResource(R.string.accept_permission)
+                                    )
+                                }
+                            })
+                    }
+                }
+            }
+        }
+    }
+}
+
+fun isLocationEnabledCheck() {
+
+}
+
+@Composable
+fun enableLocation() {
+    val activity: Context = LocalContext.current as Activity
+    val intent = Intent(Settings.ACTION_LOCATION_SOURCE_SETTINGS);
+    startActivity(activity, intent, null)
+}
```

Comment:
This is not used anywhere, but it should be used to help a user who denied the location permission to re-enable it in settings.

Reply:
This would be used after isLocationEnabled() is called so it prompts the user to turn on the gps/location if it was previously turned off.